src/pages/Home/Home.page.jsx

```diff
@@ -1,38 +1,82 @@
-import React, { useRef } from 'react';
-import { Link, useHistory } from 'react-router-dom';
-
-import { useAuth } from '../../providers/Auth';
-import './Home.styles.css';
+import React, { useContext, useEffect } from 'react';
+import useFetch from '../../hooks/useFetch';
+// import demoData from '../../utils/response.json';
+import YoutubeListItem from '../../components/YoutubeList/YoutubeListItem';
+import GlobalContext from '../../state/GlobalContext';
 
 function HomePage() {
-  const history = useHistory();
-  const sectionRef = useRef(null);
-  const { authenticated, logout } = useAuth();
+  const { query, responseList, setResponseList } = useContext(GlobalContext);
+  const list = [];
+  const params = {
+    part: 'snippet',
+    key: process.env.REACT_APP_YOUTUBE_API_KEY,
+    maxResults: 35,
+    type: 'video',
+    videoSyndicated: 'true',
+  };
+
+  const url =
+    query &&
+    `https://youtube.googleapis.com/youtube/v3/search?part=${params.part}&key=${params.key}&maxResults=${params.maxResults}&type=${params.type}&videoSyndicated=${params.videoSyndicated}&q=${query}`;
+
+  const { loading, data } = useFetch(url);
 
-  function deAuthenticate(event) {
-    event.preventDefault();
-    logout();
-    history.push('/');
+  useEffect(() => {
+    setResponseList(data.items);
+  });
+
+  if (responseList === undefined) {
+    list.push(<h1 key="welcome">Welcome Stranger!</h1>);
+  }
+  if (responseList) {
+    responseList.forEach((element) => {
+      list.push(
+        <YoutubeListItem
+          key={element.id.videoId}
+          title={element.snippet.title}
+          description={element.snippet.description}
+          imageUrl={element.snippet.thumbnails.medium.url}
+          videoId={element.id.videoId}
+        />
+      );
+    });
   }
 
   return (
-    <section className="homepage" ref={sectionRef}>
-      <h1>Hello stranger!</h1>
-      {authenticated ? (
-        <>
-          <h2>Good to have you back</h2>
-          <span>
-            <Link to="/" onClick={deAuthenticate}>
-              ← logout
-            </Link>
-            <span className="separator" />
-            <Link to="/secret">show me something cool →</Link>
-          </span>
-        </>
+    <div>
+      {loading ? (
+        <div className="loadingio-spinner-spin-u1zq1d0hnx">
+          <div className="ldio-zsz4poalow">
+            <div>
+              <div />
+            </div>
+            <div>
+              <div />
+            </div>
+            <div>
+              <div />
+            </div>
+            <div>
+              <div />
+            </div>
+            <div>
+              <div />
+            </div>
+            <div>
+              <div />
+            </div>
+            <div>
+              <div />
+            </div>
+            <div>
+              <div />
+            </div>
+          </div>
+        </div>
```

Comment:
Are all these divs part of the spinner? Maybe you could try to create a reusable component

Reply:
I take the chance and change it, I modified the spinner, now its an styled component and as a reusable component.